eternalcombat-api/src/main/java/com/eternalcode/combat/region/WorldGuardRegionProvider.java

```diff
@@ -1,69 +1,76 @@
-    }
-
-    @Override
-    public Location getRegionCenter(Location location) {
-        RegionContainer regionContainer = WorldGuard.getInstance().getPlatform().getRegionContainer();
-        RegionQuery regionQuery = regionContainer.createQuery();
+    public Optional<Region> getRegion(Location location) {
+        RegionQuery regionQuery = this.regionContainer.createQuery();
         ApplicableRegionSet applicableRegions = regionQuery.getApplicableRegions(BukkitAdapter.adapt(location));
 
-        double minX = 0;
-        double maxX = 0;
-        double minZ = 0;
-        double maxZ = 0;
-
         for (ProtectedRegion region : applicableRegions.getRegions()) {
-            BlockVector3 min = region.getMinimumPoint();
-            BlockVector3 max = region.getMaximumPoint();
-
-            if (min.getX() < minX) {
-                minX = min.getX();
-            }
-            if (max.getX() > maxX) {
-                maxX = max.getX();
-            }
-            if (min.getZ() < minZ) {
-                minZ = min.getZ();
-            }
-            if (max.getZ() > maxZ) {
-                maxZ = max.getZ();
+            if (!isCombatRegion(region)) {
+                continue;
             }
+
+            return Optional.of(new RegionImpl(location, region));
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean isCombatRegion(ProtectedRegion region) {
+        if (this.regions.contains(region.getId())) {
+            return true;
         }
 
-        double x = (maxX - minX) / 2 + minX;
-        double z = (maxZ - minZ) / 2 + minZ;
+        if (this.pluginConfig.settings.shouldPreventPvpRegions) {
+            StateFlag.State flag = region.getFlag(Flags.PVP);
+
+            if (flag != null) {
+                return flag.equals(StateFlag.State.DENY);
+            }
+        }
 
-        return new Location(location.getWorld(), x, location.getY(), z);
+        return false;
     }
 
-    private boolean isInCombatRegion(Location location, String regionName) {
-        RegionContainer regionContainer = WorldGuard.getInstance().getPlatform().getRegionContainer();
-        RegionQuery regionQuery = regionContainer.createQuery();
-        ApplicableRegionSet applicableRegions = regionQuery.getApplicableRegions(BukkitAdapter.adapt(location));
+    private record RegionImpl(Location contextLocation, ProtectedRegion region) implements Region {
+        @Override
+        public Location getCenter() {
+            BlockVector3 min = region.getMinimumPoint();
+            BlockVector3 max = region.getMaximumPoint();
 
-        return applicableRegions.getRegions().stream().anyMatch(region -> region.getId().equalsIgnoreCase(regionName));
+            double x = (double) (min.getX() + max.getX()) / 2;
+            double z = (double) (min.getZ() + max.getZ()) / 2;
+
+            return new Location(contextLocation.getWorld(), x, contextLocation.getY(), z);
+        }
     }
```

Comment:
Can you move it outside this file?

Reply:
This is very internal record. It also comes with WorldGuard enabled